sound/soc/sof/intel/hda-loader.c

```diff
@@ -32,11 +33,6 @@ static int cl_stream_prepare(struct snd_sof_dev *sdev, unsigned int format,
 	struct pci_dev *pci = to_pci_dev(sdev->dev);
 	int ret;
 
-	if (direction != SNDRV_PCM_STREAM_PLAYBACK) {
-		dev_err(sdev->dev, "error: code loading DMA is playback only\n");
-		return -EINVAL;
-	}
-
```

Comment:
Humm, it'd be useful to keep this check for regular code loading.

maybe rename this as cl_stream_prepare_common() and keep the direction check in cl_stream_prepare before calling cl_stream_prepare_common()

Reply:
@plbossart this is a static funtion called for 2 streams one playback and capture. I feel that keeping this check only for playback would add unncessary code. If we do this for playback, we should do it for capture as well and then it would be duplicating the same things in 2 separate functions. But if you feel strongly about this, I can split the function and add the check. Let me know!

Reply:
well if someone screws-up on the direction nothing will work anyways, so not sure if an error message will really be useful. let's drop my comment